code/modules/projectiles/projectile/energy.dm

```diff
@@ -20,7 +24,28 @@
 	//Damage will be handled on the MOB side, to prevent window shattering.
 
 /obj/item/projectile/energy/electrode/on_hit(var/atom/target, var/blocked = 0)
+	var/deflected = NO_DEFLECTION //0 for not deflected, 1 for partially, 2 for totally.
+	if(ishuman(target)) //have to do this stuff before we call ..(), but I want the messages to come in the right order. So the deflected var.
+		var/mob/living/carbon/human/H = target//typecast
+		var/protection = H.getarmor(def_zone, "shock") //checks for taser resistance of worn armor
+		if(prob(protection)) //chance to completely negate taser hit
+			stun = 0
+			weaken = 0
+			stutter = 0
+			jitter = 0
+			deflected = FULL_DEFLECTION
+		else if(prob(protection)) //chance for a glancing blow if first check fails
+			stun = 0
+			weaken = 0
+			H.apply_effect(stamina, 50, 0) //stamina damage, still better than being stunned.

```

Comment:
I'm not sure if your usage of this is correct. You probably want `apply_damage()`